contracts/bond/CollateralWhitelist.sol

```diff
@@ -67,22 +72,29 @@ abstract contract CollateralWhitelist is Initializable {
 
         string memory symbol = IERC20MetadataUpgradeable(erc20CollateralTokens)
             .symbol();
-        require(isCollateralWhitelisted(symbol), "Whitelist: not whitelisted");
         require(
-            _whitelist[symbol] != erc20CollateralTokens,
-            "Whitelist: identical address"
+            isCollateralWhitelisted(erc20CollateralTokens),
+            "Whitelist: not whitelisted"
+        );
+        bytes32 storedSymbolHash = keccak256(
+            abi.encodePacked(_symbols[erc20CollateralTokens])
```

Comment:
Getting an error trying to compare two strings in memory lol!

Reply:
I agree with Solidity on this, comparing string is the work of heretics. Why would you ever need anything other than `bytes[]` 😁

Reply:
Is that going to be an expensive gas operation?

Reply:
Excluding the SLOAD (which needs to happen anyway for comparison?) it costs 1112 gas with whatever the compiler optimisation config is atm, which seems expensive. 
Changing to `abi.encode()` vs `abi.encodePacked()` uses another 100 or so gas.